arbitrator/prover/src/machine.rs

```diff
@@ -531,6 +542,36 @@ impl Module {
             memory_store_internal_type,
         ));
 
+        //we're adding two internal functions here which implement in WAVM the
+        //memory.fill and memory.copy instructions in WASM
+        let types = &BULK_MEM_WASM.types;
+        assert_eq!(types.len(), 1);
+        let func_type = types.get(0).unwrap();
+        let [memset, memcpy] = [0, 1].map(|i| {
+            let code = BULK_MEM_WASM.codes.get(i).unwrap();
+            Function::new(
+                &code.locals,
+                |code_vec| {
+                    wasm_to_wavm(
+                        &code.expr,
+                        code_vec,
+                        floating_point_impls,
+                        &Vec::new(), //only used for Call instrs, which there are none
+                        types,
+                        0,
+                        internals_offset,
+                    )
+                },
+                func_type.clone(),
+                types,
+            )
+            .unwrap()
+        });
+        //internals offset + 4
+        code.push(memset);
+        //internals offset + 5
+        code.push(memcpy);
```

Comment:
You should also add entries to `func_types` for these functions

It may make sense to have the above array map be a closure that you call twice